tests/e2e_logging.py

```diff
@@ -1382,16 +1428,23 @@ def test_random_receipts(
                     verify_receipt(
                         receipt,
                         network.cert,
-                        not lts,
                         claims=additional_seqnos.get(s),
                         generic=True,
+                        skip_endorsement_check=lts,
                     )
                     if s == max_seqno:
                         # Always a signature receipt
                         assert receipt["proof"] == [], receipt
                     break
                 elif rc.status_code == http.HTTPStatus.ACCEPTED:
                     time.sleep(0.5)
+                elif (
+                    lts
+                    and rc.status_code == http.HTTPStatus.INTERNAL_SERVER_ERROR
+                    and "is in a pre-2.0 part of the ledger"
+                    in rc.body.json()["error"]["message"]
```

Comment:
@achamayou The endpoint currently throws exceptions when it can't find old service identities and when node certificates are missing, so I catch them here in the test scripts. This may break behavior which is only noticed after an upgrade to 2.0. Alternatively, we could either just ignore the errors and deliver empty endorsements (pretty much 1.x behaviour). In that case I would like to add some kind of warning message though. Do we have warning headers for HTTP replies or something alike?

Reply:
I think it's fine not to return endorsements for pre-2.0 transactions, and it needn't be an error. We very rarely retroactively add features, there's not a particularly strong reason to expect this one to be different?

We could set http headers, but there's no standard for warning, and it's very safe to assume it would be ignored.

I think it's particularly ok in the case of receipts, because there's a strong assumption they're checked by the client quite quickly (waiting until audit time to do so is extremely obviously flawed).